solution/index.js

```diff
@@ -0,0 +1,317 @@
+    if (target === toDoButtonEl || 
+        target === inProgressElButtonEl || 
+        target === doneButtonEl ){
+            
+        const section = target.closest("section");                 // find the section of the add button event
+        let inputTextValue = section.querySelector("input").value; // take the value of the input
+        if (inputTextValue === ""){                                // input no value ---> alert
+            return alert("You did not enter a task")
+        }
+        const ul = section.querySelector("ul");
+        const newTask = createElement("li", inputTextValue, ["task"], 
+                    {draggable:"true", contenteditable:true},
+                    {click: editTask, mouseover: switchTasks, dragstart:dragStart})
+        ul.insertBefore(newTask, ul.firstChild);                    // add before first child
+        addTasksToLocal(target, inputTextValue)                     // update to local
+        section.querySelector("input").value = "";                  // reset input value
+    }
+}
+
+// update the task to local
+function addTasksToLocal(target, newTask){
+
+    const section = target.closest("section");
+    let PropObj = section.id;
+    if (target === toDoButtonEl){
+        data[PropObj].unshift(newTask)
+    }
+    if (target === inProgressElButtonEl){
+        data[PropObj].unshift(newTask)
+    }
+    if (target === doneButtonEl){
+        data[PropObj].unshift(newTask)
+    }
+    localStorage.setItem("tasks",JSON.stringify(data))
+}
+
+
+// edit task function ---> edit the task if i dblclick
+function editTask (event){
+    const liEl = event.target;
+    liEl.style.backgroundColor = "whitesmoke";                // if i edit the background is change
+    const oldValue = liEl.textContent;                        // remove the value data letter
+    
+    liEl.addEventListener("blur", Blur);
+    function Blur (){
+        const newValue = liEl.textContent;                    // add the correct value to data
+        const section = liEl.closest("section");
+        if (section === null) return;
+        let PropObj = section.id;
+        
+        if (liEl.textContent === ""){                         // if text is empty ---> remove and alert
+            alert("you delete your task");
+            liEl.remove(); 
+            let index = data[PropObj].indexOf(oldValue);
+            data[PropObj].splice(index,1);
+        }
+        else{                                                 // else ---> uptade the task in data
+            let index = data[PropObj].indexOf(oldValue);
+            data[PropObj].splice(index,1,newValue);
+            console.log(data)
+        }
+        liEl.style.backgroundColor="rgb(192, 236, 172)";      // change the background agian
+        localStorage.setItem("tasks",JSON.stringify(data))    // change the local storage to data
+    }
+}
+
+// function switchTasks ---> switchTasks
+let liEl;
+function switchTasks(e){
+    liEl = e.target;
```

Comment:
Why do you need this global variable if `e.target` is accessible inside the function?